accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts

```diff
@@ -931,47 +931,48 @@ let a11yNls = {
-    },
-    // JCH - DONE
-    "HAAC_Combobox_Autocomplete_Invalid": {
-        0: "The 'aria-autocomplete' attribute value for a combobox text input element must not be \"inline\"",
-        "Pass_0": "Rule Passed",
-        "Fail_1": "The 'aria-autocomplete' attribute value of \"inline\" is not valid for the combobox"
-    },
-    // JCH - DONE
-    "HAAC_Combobox_Expanded": {
-        0: "The 'aria-owns' or the 'aria-controls' attribute of the expanded combobox must reference a valid popup 'id' value",
-        "Pass_0": "Rule Passed",
-        "Fail_1": "The 'aria-owns' or the 'aria-controls' attribute of the expanded combobox does not reference a valid popup 'id' value"
-    },
-    // JCH - DONE
-    "HAAC_Combobox_Popup": {
-        0: "The value of the combobox 'aria-haspopup' attribute must match the 'role' value of the popup element",
-        "Pass_0": "Rule Passed",
-        "Fail_1": "The value of the combobox 'aria-haspopup' attribute does not match the 'role' value of the popup element"
+    "combobox_version": {
+        0: "The combobox design pattern must be valid for WAI-ARIA 1.2",
+        "Pass_1.0": "The combobox design pattern is detected as WAI-ARIA 1.0, which is allowed by WAI-ARIA 1.2",
+        "Fail_1.1": "The combobox design pattern is detected as WAI-ARIA 1.1, which is not allowed by WAI-ARIA 1.2",
+        "Pass_1.2": "The combobox design pattern is detected as WAI-ARIA 1.2"
+    },
+    "combobox_popup_reference": {
+        0: "The 'aria-controls' (for WAI-ARIA 1.2) or the 'aria-owns' (for WAI-ARIA 1.0) attribute of the expanded combobox must reference a valid popup 'id' value",
+        "Pass_expanded": "The combobox popup referenced by 'aria-controls' (WAI-ARIA 1.2) or 'aria-owns' (WAI-ARIA 1.0) exists and is visible",
+        "Pass_collapsed": "The combobox popup is not referenced by 'aria-controls' (WAI-ARIA 1.2) or 'aria-owns' (WAI-ARIA 1.0) or is hidden",
+        "Fail_1.0_missing_owns": "The 'aria-owns' attribute of the expanded combobox is missing",
+        "Fail_1.2_missing_controls": "The 'aria-controls' attribute of the expanded combobox is missing",
+        "Fail_1.0_popup_reference_missing": "The 'aria-owns' attribute \"{0}\" of the expanded combobox does not reference a valid popup 'id' value",
+        "Fail_1.2_popup_reference_missing": "The 'aria-controls' attribute \"{0}\" of the expanded combobox does not reference a valid popup 'id' value",
+        "Fail_combobox_expanded_hidden": "The combobox 'aria-expanded' attribute is true, but the combobox popup is not visible",
+        "Fail_combobox_collapsed_visible": "The combobox 'aria-expanded' attribute is false, but the combobox popup is visible"
+    },
+    "combobox_haspopup": {
+        0: "The combobox attribute 'aria-haspopup' value must be appropriate for the role of the element referenced by `aria-controls` (WAI-ARIA 1.2) or `aria-owns` (WAI-ARIA 1.0)",
```

Comment:
If I remember correctly, we cannot use the backquote in these messages, so the quotes around aria-controls and aria-owns might need to be changed.